src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -2665,6 +2664,93 @@ let updateTopLevelMetadata (arcFiles: ArcFiles) =
         }
     )
 
+let fillSelectedBuildingBlocksWithOntologyAnnotation (ontologyAnnotation: OntologyAnnotation) =
+    Excel.run(fun context ->
+        promise {
+            let! result = tryGetActiveAnnotationTable context
+
+            match result with
+            | Some excelTable ->
+
+                let! selectedBuildingBlock = getSelectedBuildingBlock excelTable context
+                let selectedRange = context.workbook.getSelectedRange().load(U2.Case2 (ResizeArray[|"rowCount"; "rowIndex"|]))
+
+                let tableRange = excelTable.getRange()
+                let _ = tableRange.load(U2.Case2 (ResizeArray[|"values"|]))
+
+                do! context.sync().``then``(fun _ -> ())
+
+                let tableValues =
+                    tableRange.values
+                    |> Array.ofSeq
+                    |> Array.map (fun row ->
+                        row
+                        |> Array.ofSeq
+                        |> Array.map (fun column ->
+                            column
+                            |> Option.map string
+                            |> Option.defaultValue ""
+                            |> (fun s -> s.TrimEnd())
+                        )
+                    )
+
+                let tableHeaders = tableValues.[0]
+                let firstRow = selectedRange.rowIndex
+                let rowCount = selectedRange.rowCount
+
+                if firstRow = 0 && rowCount = 1 then return [InteropLogging.Msg.create InteropLogging.Error "You cannot fill the headers of an annotation table with terms!"]
```

Comment:
Maybe do:

1. Unspecialized insert if not in annotation table
2. If in annotation table do:
    1. Get selected building block
    2. Check building block type
    3. Insert accordingly

Reply:
Should be fixed